tools/gfx/vulkan/render-vk.cpp

```diff
@@ -5221,101 +5222,108 @@ VkPipelineShaderStageCreateInfo VKDevice::compileEntryPoint(
+            m_api.vkEnumerateInstanceLayerProperties(&layerCount, availableLayers.getBuffer());
+
+            const char* layerNames[] = { nullptr };
+            for (auto& layer : availableLayers)
+            {
+                if (strncmp(
                     layer.layerName,
                     "VK_LAYER_KHRONOS_validation",
                     sizeof("VK_LAYER_KHRONOS_validation")) == 0)
-            {
-                layerNames[0] = "VK_LAYER_KHRONOS_validation";
-                break;
+                {
+                    layerNames[0] = "VK_LAYER_KHRONOS_validation";
+                    break;
+                }
             }
-        }
-        // On older drivers, only "VK_LAYER_LUNARG_standard_validation" exists,
-        // so we try to use it if we can't find "VK_LAYER_KHRONOS_validation".
-        if (!layerNames[0])
-        {
-            for (auto& layer : availableLayers)
+            // On older drivers, only "VK_LAYER_LUNARG_standard_validation" exists,
+            // so we try to use it if we can't find "VK_LAYER_KHRONOS_validation".
+            if (!layerNames[0])
             {
-                if (strncmp(
+                for (auto& layer : availableLayers)
+                {
+                    if (strncmp(
                         layer.layerName,
                         "VK_LAYER_LUNARG_standard_validation",
                         sizeof("VK_LAYER_LUNARG_standard_validation")) == 0)
-                {
-                    layerNames[0] = "VK_LAYER_LUNARG_standard_validation";
-                    break;
+                    {
+                        layerNames[0] = "VK_LAYER_LUNARG_standard_validation";
+                        break;
+                    }
                 }
             }
+            if (layerNames[0])
+            {
+                instanceCreateInfo.enabledLayerCount = SLANG_COUNT_OF(layerNames);
+                instanceCreateInfo.ppEnabledLayerNames = layerNames;
+            }
         }
-        if (layerNames[0])
+        uint32_t apiVersionsToTry[] = { VK_API_VERSION_1_2, VK_API_VERSION_1_1, VK_API_VERSION_1_0 };
+        for (auto apiVersion : apiVersionsToTry)
         {
-            instanceCreateInfo.enabledLayerCount = SLANG_COUNT_OF(layerNames);
-            instanceCreateInfo.ppEnabledLayerNames = layerNames;
+            applicationInfo.apiVersion = apiVersion;
+            if (m_api.vkCreateInstance(&instanceCreateInfo, nullptr, &instance) == VK_SUCCESS)
+            {
+                break;
+            }
         }
     }
-    uint32_t apiVersionsToTry[] = {VK_API_VERSION_1_2, VK_API_VERSION_1_1, VK_API_VERSION_1_0};
-    for (auto apiVersion : apiVersionsToTry)
+    else
     {
-        applicationInfo.apiVersion = apiVersion;
-        if (m_api.vkCreateInstance(&instanceCreateInfo, nullptr, &instance) == VK_SUCCESS)
-        {
-            break;
-        }
+        instance = *(VkInstance*)handles.getVkInstance();
```

Comment:
I don't think we need the indirection here. The user is supposed to give us an VkInstance represented in a uint64. So this just need to be (VkInstance)handles.getVkInstance().

Reply:
Oh, right. Fixed.